clients/bmp_client.go

```diff
@@ -23,8 +24,19 @@ type bmpClient struct {
 
 func NewBmpClient(username, password, url string, hClient softlayer.HttpClient, configPath string) *bmpClient {
 	var httpClient softlayer.HttpClient
+
+	useHttps := false
+	if url != "" {
```

Comment:
Actually I'm intended to remove the "http://" in the url here, so the http client can concat the right url to call the server, not just to check whether it is using https or not. So I think it's better to keep it here:)